File: src/main/java/fi/metatavu/metaform/server/email/EmailTemplateSource.java
package fi.metatavu.metaform.server.email;

import java.util.UUID;

import org.apache.commons.lang3.StringUtils;

/**
 * Freemarker template source
 * 
 * @author Antti Leppä
 * @author Heikki Kurhinen
 */
public enum EmailTemplateSource {
  
  EMAIL_CONTENT ("email-content-"),
  EMAIL_SUBJECT ("email-subject-");
  
  EmailTemplateSource(String prefix) {
    this.prefix = prefix;
  }
  
  /**
   * Resolve source from template name
   * 
   * @param name name
   * @return source
   */
  public static EmailTemplateSource resolve(String name) {
    for (EmailTemplateSource templateSource : values()) {
      if (StringUtils.startsWith(name, templateSource.prefix)) {
        return templateSource;
      }
    }
    
    return null;
  }
  
  /**
   * Returns prefix
   * 
   * @return prefix
   */
  public String getPrefix() {
    return prefix;
  }

  /**
   * Returns template name for id
   * 
   * @param id id
   * @return template name for id
   */
  public String getName(UUID id) {
    return String.format("%s%s", getPrefix(), id);
  }

  private String prefix;
  
}

File: src/main/kotlin/fi/metatavu/metaform/server/email/EmailTemplateSource.kt
package fi.metatavu.metaform.server.email

import org.apache.commons.lang3.StringUtils
import java.util.*

/**
 * Freemarker template source
 *
 * @author Antti Leppä
 * @author Heikki Kurhinen
 */
enum class EmailTemplateSource(val prefix: String) {

    EMAIL_CONTENT("email-content-"),
    EMAIL_SUBJECT("email-subject-");

    /**
     * Returns template name for id
     *
     * @param id id
     * @return template name for id
     */
    fun getName(id: UUID): String {
        return String.format("%s%s", prefix, id)
    }

    companion object {
        /**
         * Resolve source from template name
         *
         * @param name name
         * @return source
         */
        fun resolve(name: String): EmailTemplateSource? {
            return values().firstOrNull() { templateSource ->
                StringUtils.startsWith(name, templateSource.prefix)
            }
        }
    }
}